spec/entities/base_spec.rb

```diff
@@ -113,103 +187,181 @@ class Child < KalibroClient::Entities::Base; end
     end
   end
 
-  describe 'save' do
-    context "when it is not persisted" do
-      context "when it doesn't have the method id=" do
-        before :each do
-          KalibroClient::Entities::Base.
-            expects(:request).
-            with('', {base: {}}, :post, '').returns({"base" => {'id' => 42, 'kalibro_errors' => []}})
+  shared_examples 'persistence method' do |method_name, http_method, has_id = true|
+    before :each do
+      subject.id = 42 if has_id
+    end
+
+    let(:url) { has_id ? ':id' : '' }
+    let(:params) { has_id ? has_entry(id: 42) : anything }
+
+    context 'when a record does not exist with given id' do
+      before :each do
+        subject.class.expects(:request).with(url, params, http_method, '').
+          raises(KalibroClient::Errors::RecordNotFound)
+      end
+
+      it 'should raise a RecordNotFound error' do
```

Comment:
Using `should` for spec descriptions has been deprecated. Instead, `is expected to` fits better to the code.